src/main.rs

```diff
@@ -58,6 +58,42 @@ use errors::*;
 struct Options<'a> {
     true_color: bool,
     language: Option<&'a str>,
+    interactive_terminal: bool,
+}
+
+enum Shell<'a> {
```

Comment:
Very cool!

Maybe call this `OutputTerminal` instead of `Shell`?

Reply:
Another thought:

Call this `OutputType` and name the variants `Pager` and `Stdout`? The link between interactive=>pager, non-interactive=>stdout happens somewhere else.